src/crlf.c

```diff
@@ -138,8 +138,8 @@ static int crlf_apply_to_odb(
 		if (git_buf_text_gather_stats(&stats, from, false))
 			return GIT_PASSTHROUGH;
 
-		/* If there are no CR characters to filter out, then just pass */
-		if (!stats.cr)
+		/* If there are no CR characters to filter out and CrLf is not set to "true", then just pass */
+		if (!stats.cr && ca->auto_crlf != GIT_AUTO_CRLF_TRUE)
```

Comment:
This patch perfectly works on *nix and the clar tests however not on Win32, as auto_crlf is not properly initialized based on core.autocrlf :(